### PR TITLE
[ENG-1190] New shortcuts system

### PR DESCRIPTION
**Pull request by ameer2468**
This PR makes extending and adding shortcuts easier, now they are all unified in one place, making them easy to modify aswell.

- Added duplicate shortcut
- Added additional keybinds to the keybindings page.
- 'Cut' in Edit menu **(MacOS)**
- Context menu, corrected the Delete symbol for windows.
- Removed un-necessary shortcuts code from **Quick Preview**
- If Quickpreview is open, do not open the folder within Spacedrive when you press Enter.

### CODE REVIEW

**Comment by utkubakir** [comment]
`useShortcut` can receive an array of key (shortcut) and function (handler) objects. This way we would only have 1 hook imported and not several (talking about `useKeys` here)

File: interface/hooks/useShortcut.ts [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+
+import { OperatingSystem } from '../util/Platform';
+import { useOperatingSystem } from './useOperatingSystem';
+
+const state = {
+	gridView: {
+		keys: {
+			macOS: ['Meta', '1'],
+			all: ['Control', '1']
+		}
```
Comment (utkubakir) [inline interface/hooks/useShortcut.ts:12]: have a description, label, category, or something similar so our shortcuts page can export and display this. There is no need to duplicate data which we will definitely forget to maintain

File: interface/hooks/useShortcut.ts [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+
+export const useShortcut = (shortcut: shortcutKeys, func: (e: KeyboardEvent) => void) => {
+	const os = useOperatingSystem();
+	const shortcutsStore = getShortcutsStore();
+	const shortcutKeys =
+		shortcutsStore[shortcut].keys[os as osKeys] || shortcutsStore[shortcut].keys.all;
+
+	useKeys(shortcutKeys, (e) => {
+		func(e);
+	});
```
[the remark this answers is not in the document]
Comment (ameer2468) [reply interface/hooks/useShortcut.ts:211]: My bad, i actually did that at first then undo'd.

**Review by brxken128 — APPROVED** [review]
Very happy now that we have one central place for these haha

**Pull request merged**